shared/plugin_processor.go

```diff
@@ -0,0 +1,45 @@
+package shared
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+)
+
+type AttributedObservation[ObservationType any] struct {
+	OracleID    commontypes.OracleID
+	Observation ObservationType
+}
+
+// PluginProcessor This interface is to create multiple SubPlugins/Processors under OCR plugin.
+// The idea is to separate the logic of different types of observations and outcomes into separate processors.
+// This makes it easier to manage and test the logic of each type of observation/outcome without affecting each others
+// Some of them will implement state machines (e.g. merkleroot),
+// others might implement simpler logic. (e.g. tokenprices, chainfee)
+// The OCR plugin becomes a coordinator/collector of these SubPlugins.
+// Example Pseudo code:
+//
+//	OCRPlugin {
+//	  nodeID
+//	  merkleRootsProcessor
+//	  tokenPriceProcessor
+//	  chainFeeProcessor
+//	  ...
+//	}
+//
+//	OCRPlugin.Observation {
+//	  mObs := merkleProcessor.Observation
+//	  tObs := tokenProcessor.Observation
+//	  fObs := feeProcessor.Observation
+//	  return Observation{mObs, tObs, fObs}
+//	}
```

Comment:
Godoc supports code blocks.

[Documentation](https://go.dev/doc/comment).
* [relevant section](https://go.dev/doc/comment#code): A code block is a span of indented or blank lines not starting with a bullet list marker or numbered list marker.
```suggestion
//		 type OCRPlugin struct {
//	 	  merkleRootsProcessor
//		   tokenPriceProcessor
//	 	   chainFeeProcessor
//		 }
//
//      // Observation excludes error handling for brevity.
// 	 func (p *OCRPlugin) Observation() ocrtype.Observation {
//	 	 	 return aggregatedObservations{
//            merkleRoots: p.merkleRootsProcessor.Observe(...)
//            tokenPrices: p.tokenPriceProcessor.Observe(...)
//            chainFees: p.chainFeeProcessor.Observe(...)
//          }
//	 	 }
```